code/modules/mob/living/carbon/carbon_update_icons.dm

```diff
@@ -237,27 +237,44 @@
 	update_wound_overlays()
 	var/list/needs_update = list()
 	var/limb_count_update = FALSE
+	var/obj/item/bodypart/l_leg/left_leg
+	var/obj/item/bodypart/r_leg/right_leg
+	var/old_left_leg_key
 	for(var/obj/item/bodypart/limb as anything in bodyparts)
 		limb.update_limb(is_creating = update_limb_data) //Update limb actually doesn't do much, get_limb_icon is the cpu eater.
+
+		if(limb.body_zone == BODY_ZONE_R_LEG)
+			right_leg = limb
+			continue // Legs are handled separately
+
 		var/old_key = icon_render_keys?[limb.body_zone] //Checks the mob's icon render key list for the bodypart
 		icon_render_keys[limb.body_zone] = (limb.is_husked) ? limb.generate_husk_key().Join() : limb.generate_icon_key().Join() //Generates a key for the current bodypart
+
+		if(limb.body_zone == BODY_ZONE_L_LEG)
+			left_leg = limb
+			old_left_leg_key = old_key
+			continue // Legs are handled separately
```

Comment:
why is this done like this? is the order important?

Reply:
Yes, it's to avoid regenerating a icon_key unnecessarily, because we have to do it later on for he right leg, once he left leg has been handled.